sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryStrategy.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import java.time.Duration;
+
+/**
+ * The interface for determining the retry strategy used in {@link RetryPolicy}.
+ */
+public interface RetryStrategy {
```

Comment:
Could this interface have an apply retry/attempt retry added to it? This way the strategy implementation is the only thing the RetryPolicy needs to run.

Reply:
I have added a default method called `shouldRetry()` to the interface which will make it more flexible while the RetryPolicy is still in control of actually attempting the retry.